nixos/modules/services/networking/adguardhome.nix

```diff
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.adguardhome;
+
+  args = concatStringsSep " " ([
+    "--no-check-update"
+    "--config ${cfg.confFile}"
+    "--work-dir ${cfg.workDir}"
+    "--host ${cfg.host}"
+    "--port ${toString cfg.port}"
+  ] ++ optional (cfg.logFile != null) "--logfile ${cfg.logFile}"
+  ++ optional (cfg.pidFile != null) "--pidfile ${cfg.pidFile}"
+  ++ optional cfg.verbose "--verbose");
+
+in
+{
+  options.services.adguardhome = with types; {
+    enable = mkEnableOption "AdGuard Home network-wide ad blocker";
+
+    confFile = mkOption {
+      default = "/etc/AdGuardHome.yaml";
+      type = path;
+      description = ''
+        Path to the config file.
+      '';
+    };
+
+    workDir = mkOption {
+      default = "/var/lib/AdGuardHome";
+      type = path;
+      description = ''
+        Path to the working directory.
+      '';
+    };
+
+    host = mkOption {
+      default = "0.0.0.0";
+      type = str;
+      description = ''
+        Host address to bind HTTP server on.
+      '';
+    };
+
+    port = mkOption {
+      default = 3000;
+      type = ints.u16;
+      description = ''
+        Port to serve HTTP pages on.
+      '';
+    };
+
+    logFile = mkOption {
+      default = "syslog";
+      type = nullOr (either path (enum [ "syslog" ]));
+      description = ''
+        Path to log file. If null: write to stdout; if 'syslog': write to system log.
+      '';
+    };
+
+    pidFile = mkOption {
+      default = "/run/AdGuardHome.pid";
```

Comment:
```suggestion
      default = "/run/AdGuardHome/AdGuardHome.pid";
```

Reply:
This fails with a "no such file or directory error", looks like the directory must exist. Is leaving it in `/run` or should there be some mechanism to ensure `/run/AdGuardHome` exists?

Reply:
I am not sure what the correct way is to create the directory but otherwise it isn't easily mountable in containers.

Reply:
The correct way is using https://www.freedesktop.org/software/systemd/man/systemd.exec.html#RuntimeDirectory=.